src/Lockfile.hpp

```diff
@@ -20,6 +20,7 @@
 
 #include "system.hpp"
 
+#include <algorithm>
```

Comment:
Suggestion: Move this to the `.cpp` file instead since it's not required by anything in the header file.